extensions/zendesk-agent-toolkit/src/ai-suggestions.tsx

```diff
@@ -0,0 +1,220 @@
+import { Action, ActionPanel, List, Detail, showToast, Toast, popToRoot, Icon } from "@raycast/api";
+import React, { useEffect, useState } from "react";
+import { MacroSuggestion } from "./ai-service";
+import { ticketMonitor } from "./ticket-monitor";
+import { createMacro } from "./zendesk";
+import { AISuggestionListItem } from "./components/common";
+
+export interface AISuggestionsProps {
+  onRefresh?: () => void;
+}
+
+const AISuggestions: React.FC<AISuggestionsProps> = ({ onRefresh }) => {
+  const [loading, setLoading] = useState(true);
+  const [suggestions, setSuggestions] = useState<MacroSuggestion[]>([]);
+
+  useEffect(() => {
+    loadSuggestions();
+  }, []);
+
+  async function loadSuggestions() {
+    setLoading(true);
+    try {
+      const newSuggestions = await ticketMonitor.checkForResolvedTickets();
+      setSuggestions(newSuggestions);
+
+      if (newSuggestions.length > 0) {
+        await showToast({
+          style: Toast.Style.Success,
+          title: `Found ${newSuggestions.length} AI macro suggestions`,
+        });
+      }
+    } catch (e) {
+      await showToast({
+        style: Toast.Style.Failure,
+        title: "Failed to load AI suggestions",
+        message: String(e),
+      });
+    } finally {
+      setLoading(false);
+    }
+  }
```

Comment:
This is a good use case for Raycast Hooks:
```suggestion
  const {isLoading, data: suggestions} = usePromise(async() => {
    const data = await ticketMonitor.checkForResolvedTickets();
    if (data.length > 0) {
        await showToast({
          style: Toast.Style.Success,
          title: `Found ${data.length} AI macro suggestions`,
        });
      }
      return data;
  }, [], {
    failureToastOptions: {
      title: "Failed to load AI suggestions"
    }
  })
```